src/common/schedulers/slurm_commands.py

```diff
@@ -85,6 +95,7 @@ def __init__(self, name, nodeaddr, nodehostname, state, partitions=None):
         self.nodehostname = nodehostname
         self.state = state
         self.partitions = partitions.strip().split(",") if partitions else None
+        self.instance = None
```

Comment:
> We always don't have instance when we initialize SlurmNode

Not necessarily, if we change the order and retrieve instances info before slurm info, we might be able to initialize SlurmNode with the corresponding instance directly.

I think it would be best to initialize all attributes of a class when we create the object. Yes we can change attributes anytime, but those changes are more difficult to track. I think it would be best if we can just initialize the object once, and not touch the attributes unless we really have to.

Reply:
The reason we are doing this is because the SlurmNode is initialize through` _parse_nodes_info`, this function is not only use in clustermgtd, but also in resume. So we don't want to parse additional parameter for ` _parse_nodes_info`